src/main/scala/scorex/crypto/authds/merkle/serialization/BatchMerkleProofSerialization.scala

```diff
@@ -0,0 +1,44 @@
+package scorex.crypto.authds.merkle.serialization
+
+import com.google.common.primitives.{Bytes, Ints}
+import scorex.crypto.authds.{EmptyByteArray, Side}
+import scorex.crypto.hash.Digest
+
+class BatchMerkleProofSerialization[D <: Digest]  {
```

Comment:
BatchMerkleProofSerialization => BatchMerkleProofSerializer

Reply:
D <: Digest => D <: Digest32 , as serialization assumes hashes of 256 bits.

Better to use DigestSize = 32 constant instead of magic value in the class ( DigestSize + 4 instead of 36), then it will be simpler to retarget for hashed of 512 bits if needed.

Reply:
Fixed, and went a bit further with the constants use to make the code more readable.